File: ros2_ws/src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode : public rclcpp::Node 
{
public:
    MyNode() : Node("cpp_test"), counter_(0) {
        RCLCPP_INFO(this->get_logger(), "Hello World");
        timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                        std::bind(&MyNode::timerCallback, this));
    }
private:
    void timerCallback() {
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
        counter_++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv); // inits ros2
    auto node = std::make_shared<MyNode>();
    // auto node = std::make_shared<rclcpp::Node>("cpp_test"); 
    // Create a shared pointer to a Node obj
    // RCLCPP_INFO(node->get_logger(), "Hello World");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
};

File: ros2_ws/src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy

from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.count = 0
        self.get_logger().info('Hello World')
        self.create_timer(1.0, callback=self.timer_callback)

    def timer_callback(self):
        self.count += 1
        self.get_logger().info(str(self.count))

def main(args=None):
    rclpy.init(args=args) # inits rclpy
    
    node = MyNode() # Creates a node object
    rclpy.spin(node=node) # Runs until keyboard interrupt
    rclpy.shutdown() # Shutdown

if __name__ == "__main__":
    main()

# ros2 pkg create <pkg_name> --build-type ament_python --dependencies rclpy
# chmod +x my_first_node.py

# never build in the src file